pkg/cluster/internal/create/actions/kubeadmjoin/join.go

```diff
@@ -211,6 +212,16 @@ func runKubeadmJoinControlPlane(
 		return errors.Wrap(err, "failed to join a control plane node with kubeadm")
 	}
 
+	// Wait for the node to be Ready
+	// TODO: remove once https://github.com/kubernetes-sigs/kind/issues/588 is fixed
+	// kubeadm join should guarantee that the cluster is ready
```

Comment:
@fabriziopandini @neolit123 I don't know if this is true 😅 , should `kubeadm join`, guarantee that?

Reply:
@aojea see my comments on the issue https://github.com/kubernetes-sigs/kind/issues/588#issuecomment-499939113

In a nutshell Kubeadm is not responsible, but it is the kubelet. Additional, it seems also that the API server does not detected properly when the etcd instance is ready

Reply:
> kubeadm join should guarantee that the cluster is ready

well kubeadm can look at pod and endpoint state, but the cluster as a whole - a bit tricky.
kinder currently waits for these pods + the node ready status, as @fabriziopandini  mentioned:
https://github.com/kubernetes/kubeadm/blob/62556834c87e34004ac84c17b2f2c68b5c4f3b22/kinder/pkg/actions/waiter.go#L32-L44

given HA join consistently does not fail using kind 0.2.0 as seen here:
https://k8s-testgrid.appspot.com/sig-cluster-lifecycle-kubeadm#kubeadm-kind-master
i'm trying to get to the bottom of the problem instead - i.e. finding a change in kind that helped the problem surface.

Reply:
kind is much faster with the latest versions, the fact that adding delays solve the problem or at least reduce them makes me think that´s tightly related to that